budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -1,247 +1,1041 @@
+                irm_snapshot=self.irm_snapshot,
+                project=project,
                 name="Project {}".format(i),
-                department="Department {}".format(i),
-                local_municipality="Local {}".format(i),
-                district_municipality="District {}".format(i),
                 province=province,
-                status=status,
-                primary_funding_source=source,
-                main_contractor="Contractor {}".format(i),
-                principle_agent="Principle Agent {}".format(i),
-                program_implementing_agent="Program Agent {}".format(i),
-                other_parties="Service Provider: XXXX{}".format(i),
+                estimated_completion_date=date(year=2020, month=1, day=1),
             )
+        ProvInfraProjectIndex().reindex()
 
-    def test_projects_per_page(self):
-        response = self.client.get(self.url, format="json")
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
+    def tearDown(self):
+        ProvInfraProjectIndex().clear()
+        self.file.close()
+
+    def test_search_homepage_correct_numbers(self):
+        selenium = self.selenium
+        selenium.get("%s%s" % (self.live_server_url, self.url))
+        num_of_projects = selenium.find_element_by_css_selector(
+            "#num-matching-projects-field"
+        ).text
+        num_of_projects = int(num_of_projects)
+        self.assertEqual(num_of_projects, 11)
+
+        num_of_items = len(
+            selenium.find_elements_by_css_selector(
+                "#result-list-container .narrow-card_wrapper"
+            )
+        )
+        self.assertEqual(num_of_projects, num_of_items)
+
+    def test_number_updated_after_search(self):
+        province = "Eastern Cape"
+        selenium = self.selenium
+        selenium.get("%s%s" % (self.live_server_url, self.url))
+        num_of_projects = selenium.find_element_by_css_selector(
+            "#num-matching-projects-field"
+        ).text
+        num_of_projects = int(num_of_projects)
+        self.assertEqual(num_of_projects, 11)
+
+        search_field = selenium.find_element_by_css_selector(
+            "#Infrastructure-Search-Input"
+        )
+        search_button = selenium.find_element_by_css_selector("#Search-Button")
+        search_field.send_keys(province)
+        search_button.click()
+        wait = WebDriverWait(selenium, self.timeout)
+        wait.until(
+            EC.text_to_be_present_in_element(
+                (By.CSS_SELECTOR, ".page-heading"),
+                u"Provincial Infrastructure Projects",
+            )
+        )
 
-        number_of_results = len(response.data["results"])
-        self.assertLessEqual(number_of_results, 20)
+        filtered_num_of_projects = selenium.find_element_by_css_selector(
+            "#num-matching-projects-field"
+        ).text
+        filtered_num_of_projects = int(filtered_num_of_projects)
+        self.assertEqual(filtered_num_of_projects, 5)
```

Comment:
this is where you want to do a wait.until - in one expression, you can say wait until #num-matching-projects-field contains "5" with the example I gave.

That takes care of waiting for javascript, finding the element, comparing the value, and raising an exception if it doesn't happen within the timeout. It also polls, so it should be pretty quick, rather than sleeping 5 seconds and then checking.